src/exception/OcppError.ts

```diff
@@ -3,6 +3,9 @@
  * Attributes name are part of the OCPP specification
  */
 export default class OCPPError extends Error {
+  code: string;
```

Comment:
Visibility private?

Reply:
It can't, it's used to properly build the error message to send: 

`messageToSend = JSON.stringify([messageType, messageId, commandParams.code ? commandParams.code : OCPPErrorType.GENERIC_ERROR, commandParams.message ? commandParams.message : '', commandParams.details ? commandParams.details : {}]);`

The code was assigning default values that was hiding the issue: the code never sent the exact error. 
That's because we still have any type usage. It avoids to catch missing but required attributes like here. I've tried to reduce it in important code paths lately like the WebSockets handling or OCPP messages handling, but that's still not enough to catch mistyping that reveal some bugs.

Reply:
Then public ? :-)
Visibility must be po

Reply:
Then public ? :-)
Visibility must be po

Reply:
I don't see where OCPPError is used in your example.
If you can access instance members in class OCPPError, make them public.
Otherwise encapsulation must be applied with getters and make instance members private.

Reply:
sendMessage() signature: 

`public async sendMessage(messageId: string, commandParams: Record<string, unknown> | OCPPError, ... `

and commandParams is used to build the message string to send. 

I can make the OCPP attributes explicitly public instead of implicitly.

Reply:
> I can make the OCPP attributes explicitly public instead of implicitly.

YA :-)
I think there is a tslint setting for that right? :-)